rules/aip0135/method_rules.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0135
+
+import (
+	"fmt"
+
+	"github.com/googleapis/api-linter/internal/utils"
+	"github.com/googleapis/api-linter/lint"
+	"github.com/jhump/protoreflect/desc"
+)
+
+// Delete messages should have a properly named Request message.
+var requestMessageName = &lint.MethodRule{
+	Name:   lint.NewRuleName("core", "0135", "request-message", "name"),
+	URI:    "https://aip.dev/135#guidance",
+	OnlyIf: isDeleteMethod,
+	LintMethod: func(m *desc.MethodDescriptor) []lint.Problem {
+		// Rule check: Establish that for methods such as `DeleteFoo`, the request
+		// message is named `DeleteFooRequest`.
+		if got, want := m.GetInputType().GetName(), m.GetName()+"Request"; got != want {
+			return []lint.Problem{{
+				Message: fmt.Sprintf(
+					"Delete RPCs should have a request message named after the RPC, such as %q.",
+					want,
+				),
+				Suggestion: want,
+				Descriptor: m,
+			}}
+		}
+
+		return nil
+	},
+}
+
+// Delete messages should use google.protobuf.Empty, google.longrunning.Operation or the resource itself as the response message
+var responseMessageName = &lint.MethodRule{
+	Name:   lint.NewRuleName("core", "0135", "response-message", "name"),
+	URI:    "https://aip.dev/135#guidance",
+	OnlyIf: isDeleteMethod,
+	LintMethod: func(m *desc.MethodDescriptor) []lint.Problem {
+		// Rule check: Establish that for methods such as `DeleteFoo`, the response
+		// message is `google.protobuf.Empty`, `google.longrunning.Operation`, or `Foo`.
+		got := m.GetOutputType().GetName()
+		want := []string{"Empty", "Operation", m.GetName()[6:]}
```

Comment:
I would appreciate any pointers on how to deal with protobuf packages here. Originally I had these as "google.protobuf.Empty" and "google.longrunning.Operation" but that resulted in test failures. Should the lint rules check the package, or just not bother and assume anything with the correct message name also has the correct package?

Reply:
Use [`MessageDescriptor.GetFullyQualifiedName()`](https://godoc.org/github.com/jhump/protoreflect/desc#MessageDescriptor.GetFullyQualifiedName)

Reply:
Done